tests/integration/src/xvm.rs

```diff
@@ -0,0 +1,334 @@
+        assert_ok!(EVM::call(
+            RuntimeOrigin::root(),
+            alith(),
+            call_wasm_payable_addr.clone(),
+            // to: 0x00a8f69d59df362b69a8d4acdb9001eb3e1b8d067b8fdaa70081aed945bde5c48c
+            // input: 0x0000002a (deposit)
+            // value: 1000000000
+            hex::decode("4012b914000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000c0000000000000000000000000000000000000000000000000000000003b9aca00000000000000000000000000000000000000000000000000000000000000002100a8f69d59df362b69a8d4acdb9001eb3e1b8d067b8fdaa70081aed945bde5c48c0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000040000002a00000000000000000000000000000000000000000000000000000000").expect("invalid call input hex"),
+            U256::from(value),
+            1_000_000,
+            U256::from(DefaultBaseFeePerGas::get()),
+            None,
+            None,
+            vec![],
+        ));
+        let recieved = Balances::free_balance(&wasm_payable_addr) - prev_wasm_payable_balance;
+        assert_eq!(recieved, value);
+    });
+}
+
+#[test]
+fn calling_evm_payable_from_wasm_works() {
+    new_test_ext().execute_with(|| {
+        let evm_payable_addr = deploy_evm_contract(EVM_PAYABLE);
+        let wasm_address = deploy_wasm_contract("call_xvm_payable");
+
+        let value = UNIT;
+
+        // TODO: after Account Unification finished, remove this mock account.
+        // It is needed now because currently the `AccountMapping` and `AddressMapping` are
+        // both one way mapping.
+        let mock_unified_wasm_account = account_id_from(h160_from(wasm_address.clone()));
+        let _ = Balances::deposit_creating(&mock_unified_wasm_account, value);
+
+        let evm_payable = evm_payable_addr.as_ref().to_vec();
+        let deposit_func = hex::decode("d0e30db0").expect("invalid deposit function hex");
+        let input = hex::decode("0000002a")
+            .expect("invalid selector hex")
+            .iter()
+            .chain(evm_payable.encode().iter())
+            .chain(deposit_func.encode().iter())
+            .cloned()
+            .collect::<Vec<_>>();
+        assert_ok!(Contracts::call(
+            RuntimeOrigin::signed(ALICE),
+            MultiAddress::Id(wasm_address.clone()),
+            value,
+            Weight::from_parts(10_000_000_000, 10 * 1024 * 1024),
+            None,
+            input,
+        ));
+
+        assert_eq!(
+            Balances::free_balance(account_id_from(evm_payable_addr)),
+            value
+        );
+
+        // TODO: after Account Unification finished, enable the wasm address balance check
+        // and remove the mock account balance check.
+        // assert_eq!(Balances::free_balance(&wasm_address), ExistentialDeposit::get());
+        assert_eq!(Balances::free_balance(&mock_unified_wasm_account), 0);
+    });
+}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The test function `calling_evm_payable_from_wasm_works` tests the cross-VM call from the Wasm VM to the EVM VM. The test deploys a Wasm contract and an EVM contract, transfers funds to the EVM contract, performs a payable call from the Wasm contract to the EVM contract's `deposit` function, and verifies the balance change. No issues found.

Overall, the changes in the `tests/integration/src/xvm.rs` file seem to be well-implemented and cover various scenarios related to cross-VM calls between the EVM and WASM virtual machines. No issues were identified in the code.


<!-- This is an auto-generated comment by OSS CodeRabbit -->